app/database.js

```diff
@@ -9,72 +9,206 @@
+		deleteDatasets: "DELETE FROM tbldatasets WHERE 1",
+		updateDatasetId : "UPDATE tbldatasets SET dataset_id=?, dataset_online=1 WHERE dataset_id=?",
+		updateItemStrength : "UPDATE  tbluser_items SET user_item_strength=? WHERE id=? ",
+		updateGUILanguage : "UPDATE tblusers SET user_language=?, user_lastedited=? WHERE user_id=?",
+		getUserDatasets : "SELECT * FROM tbldatasets WHERE dataset_user=?",
+		getUserDatasetsByModule : "SELECT * FROM tbldatasets WHERE dataset_user=? AND dataset_language=? AND dataset_subject=?",
+		getRecentDataset : "SELECT * FROM tbldatasets WHERE dataset_id=? AND ? > ?",
     getDatasetByName : "SELECT * FROM tbldatasets WHERE dataset_name=?",
-		getDatasetItems : "SELECT * FROM tblitems WHERE item_dataset_id=?" ,
+		getDatasetItems : "SELECT dataset_items FROM tbldatasets WHERE dataset_id=?" ,
+		getGUILanguages : "SELECT * FROM tbllanguages WHERE language_gui=1",
 		getUserSubjects : "SELECT * FROM tblsubjects",
-    getUser : "SELECT * FROM tblsubjects where user_id= ? ",
-    getUserbyEmail : "SELECT * FROM tblusers where user_email= ?",
-    getUserbyUsername : "SELECT * FROM tblusers where user_name= ?",
+    getUser : "SELECT * FROM tblusers WHERE user_id=? ",
+    getUserbyEmail : "SELECT * FROM tblusers WHERE user_email=?",
+    getUserbyUsername : "SELECT * FROM tblusers WHERE user_name=?",
 		getUserIdbyUsername : "SELECT user_id, user_password FROM tblusers WHERE user_name=?",
 		getUserIdbyEmail : "SELECT user_id FROM tblusers WHERE user_email=?",
     getLanguages: "SELECT * FROM tbllanguages",
-		getModules: "SELECT language_id, language_name, subject_id, subject_name FROM tbldatasets,tbllanguages,tblsubjects WHERE dataset_language=language_id AND dataset_subject=subject_id"
+		getModules: "SELECT language_id, language_name, subject_id, subject_name FROM tbldatasets,tbllanguages,tblsubjects WHERE dataset_language=language_id AND dataset_subject=subject_id AND dataset_user=?",
+		replaceUser: "REPLACE INTO tblusers VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
+		lastInsertedId: "SELECT ? FROM ? ORDER BY ? DESC LIMIT 1",
+		getSubjectByName : "SELECT * FROM tblsubjects WHERE subject_name=?"
 	};
 
+	var lastId = 0;
+
 	// Check if SQL.js has been loaded through AMD
-	var sql;
+	var sql,db,db_online;
 	if (typeof sqlite !== 'object') {
 		document.body.style.backgroundColor = 'red';
-		alert(lang("error_requirefail", "sql.js"));
+		alert("Failed to require sqlite through AMD.");
 	} else {
 		sql = sqlite;
 	}
 
-	// Initiate DB and check if there is an existing user DB
-	var read_database;
-	if (database_exists(config.constant("DATABASE_USER"))) {
-		read_database = fs.readFileSync(config.constant("DATABASE_USER"));
-	} else {
-		read_database = fs.readFileSync(config.constant("DATABASE_SLIMSTAMPEN"));
-	}
-	var db = new sql.Database(read_database);
-
 	function database_exists(path) {
 		try {
 			fs.accessSync(path, fs.F_OK);
 			return true;
 		} catch (e) {
+			console.log(e);
```

Comment:
No, because we check only with this function if the user database exists. The user database doesn't exist for first time users, and we don't want to show them that. It would be useful to check if the main database exists (added the card to Trello).